op-service/metrics/rpc_metrics.go

```diff
@@ -1,174 +1,231 @@
 package metrics
 
 import (
-	"errors"
+	"context"
 	"fmt"
 
-	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
 	RPCServerSubsystem = "rpc_server"
 	RPCClientSubsystem = "rpc_client"
-
-	BatchMethod = "<batch>"
 )
 
-type RPCClientMetricer interface {
-	RecordRPCClientRequest(method string) func(err error)
-	RecordRPCClientResponse(method string, err error)
-}
-
-type RPCServerMetricer interface {
-	RecordRPCServerRequest(method string) func()
-}
-
 type RPCMetricer interface {
-	RPCClientMetricer
-	RPCServerMetricer
+	NewRecorder(name string) rpc.Recorder
 }
 
-// RPCMetrics tracks client-only RPC metrics
-type RPCClientMetrics struct {
-	RPCClientRequestsTotal          *prometheus.CounterVec
-	RPCClientRequestDurationSeconds *prometheus.HistogramVec
-	RPCClientResponsesTotal         *prometheus.CounterVec
+// RPCMetrics tracks all the RPC metrics, both client & server.
+type RPCMetrics struct {
+	// Legacy client metrics. Do not remove labels or change settings for backward compatibility.
+	clientRequestsTotal          *prometheus.CounterVec
+	clientRequestDurationSeconds *prometheus.HistogramVec
+	clientResponsesTotal         *prometheus.CounterVec
+
+	// Legacy server metrics. Do not remove labels or change settings for backward compatibility.
+	serverRequestsTotal          *prometheus.CounterVec
+	serverRequestDurationSeconds *prometheus.HistogramVec
```

Comment:
Should these be expected to be equal to some of the New metrics? Like RequestsTotal == notificationsReceivedTotal?

Reply:
Requests and notifications are different in the json-rpc spec. The notifications are used to send back subscription events. So they don't match.